packages/@uppy/utils/src/FileProgress.ts

```diff
@@ -0,0 +1,16 @@
+interface FileProgressBase {
+  progress: number
+  uploadComplete: boolean
+  percentage: number
+  bytesTotal: number
+}
+
+export type FileProgressStarted = FileProgressBase & {
+  uploadStarted: number
+  bytesUploaded: number
+}
+export type FileProgressNotStarted = FileProgressBase & {
+  uploadStarted: null
+  bytesUploaded: false
+}
```

Comment:
this looks like it should be one type? Any reason for this? If it's really necessary, could use a comment to explain why

Reply:
Can you clarify why you think that? The two types are clearly different, why would you make them one?

Reply:
It doesn't seem correct that `uploadStarted` is a `number` is some cases and a `null` in others, probably better to have `uploadStarted?: number`? Same for `bytesUploaded`.

Reply:
It's `null` only if `bytesUploaded` is `false`. If we use `uploadStarted?: number`, TS would have no way to deduce what type to expect for `bytesUploaded`. It seems to me we would gain almost nothing (we remove 4 lines in our source file), the tradeoff being we get less precise types. Hardly worth it.

Reply:
```suggestion
// FileProgress is either started or not started. We have one type for each case:
export type FileProgressStarted = FileProgressBase & {
  uploadStarted: number
  bytesUploaded: number
}
export type FileProgressNotStarted = FileProgressBase & {
  uploadStarted: null
  bytesUploaded: false
}
```